apis/core/v1beta1/flagsourceconfiguration_types.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	SyncProviderKubernetes SyncProviderType = "kubernetes"
+	SyncProviderFilepath   SyncProviderType = "filepath"
```

Comment:
Please consider the diff highlighted in this related issue - https://github.com/open-feature/open-feature-operator/issues/433

Reply:
Do we already have a unified solution? I see that the ticket does not clearly state which option to use

Reply:
@beeme1mr @Kavindu-Dodan @thisthat which is better here? Any opinions? I suppose since we're already doing breaking changes here, it makes sense to concentrate them all here... so perhaps we should conform here in OFO to the flagd style: `file`.

Reply:
I would rather not put everything into one PR, it will make it hard to review and also hard to find the changes when we will need to locate it in the future. I would go for a separate PR, which will be a follow-up of this one. In fact one PR should not cover more that one issue.

Reply:
adapted to `file`

Reply:
cc @Kavindu-Dodan @beeme1mr